metrics/quantile.lua

```diff
@@ -270,6 +270,11 @@ function quantile.Query(stream_obj, q)
 		-- Fast path when there hasn't been enough data for a flush;
 		-- this also yields better accuracy for small sets of data.
 		local l = stream_obj.b_len
+
+		if l == 0 then
```

Comment:
It's not true. Buffer could be empty with non-empty stream. You should check stream length, not buffer

Reply:
Or number of observed values ( `s.n` ) if buffer is flushed

Reply:
Oh, I'm sorry. You're right.